### DISCUSSION TITLE
/hyndsight/workflow-in-r/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#18] [Announcements]
Workflow in R 
 https://robjhyndman.com/hyndsight/workflow-in-r/

**Comment by robjhyndman** [comment]
```header
Originally posted by patricio fuenmayor on 2009-10-16 20:50:30
````
estoy usando R para proyecciones, y sería interesante tener lo que ya
esta desarrollado.

**Comment by robjhyndman** [comment]
```header
Originally posted by gioby on 2010-04-10 08:50:43
````
and a good makefile to put it all together.

**Comment by robjhyndman** [comment]
```header
Originally posted by Brandon Bertelsen on 2011-02-02 20:00:57
````
Thanks for the outline here. Always nice to see someone else\'s process
in order to improve your own.

**Comment by robjhyndman** [comment]
```header
Originally posted by François-Philippe Dubé on 2013-03-25 15:33:49
````
Thank you for this tip. You might be interested in looking at the
ProjectTemplate
([www.projecttemplate.net](http://www.projecttemplate.net &quot;www.projecttemplate.net&quot;)),
available on CRAN. I personally find that the full project structure
proposed is too much for my needs, hence I create a minimalist structure
using one of the options in the project creation function.

**Comment by robjhyndman** [comment]
```header
Originally posted by Phillip Burger on 2014-02-03 19:14:36
````
Having good workflow is a key to increasing productivity. I\'m going to
adjust how I organize my projects after reading your post.

The other readers\' comment concerning
[http://www.projecttemplate.net](http://www.projecttemplate.net)
is also helpful.

Have you evolved how you organize your projects since your original
post?

**Comment by robjhyndman** [comment]
```header
Originally posted by jacob lindberg on 2015-07-22 09:17:42
````
I do not get the difference between Main.R and do.R as well as why to
split them up.

**Comment by robjhyndman** [comment]
```header
Originally posted by Paulo Olveira on 2016-10-23 21:10:09
````
I rather have a R markdown with snippets for loading, cleaning, defining
functions, etc. Having all in the same file make it easy to maintain,
understand, and move things around.

**Comment by El-Ayoubi-Hub** [comment]
Are you still using this method ?

**Reply by robjhyndman** [reply]
No, I'm using the targets package and workflow for most projects these days. See https://books.ropensci.org/targets/
